python/sglang/srt/entrypoints/verl_engine.py

```diff
@@ -39,11 +41,26 @@ def __init__(
         node_rank = self._tp_rank // tp_size_per_node
         first_rank_in_node = self._tp_rank % tp_size_per_node == 0
 
-        if first_rank_in_node:
+        if first_rank_in_node and "launch_server" not in kwargs:
             os.environ["SGLANG_BLOCK_NONZERO_RANK_CHILDREN"] = "0"
             self._engine = Engine(
                 **kwargs, tp_size=self._tp_size, node_rank=node_rank, nnodes=nnodes
             )
+        elif "launch_server" in kwargs and kwargs["launch_server"]:
```

Comment:
the top-level if-elif-else currently does: 

* if: for Engine, part 1
* elif: for Adapter
* else: for Engine, part 2

Thus would be great to have

```
if backend == 'engine':
  if first rank in node: self._engine = Engine()
  else: self._engine = None
elif backend == 'server':
  if rank zero: self._engine = Adapter()
  else: self._engine = None
else:
  raise error
```